front/src/common/Map/Settings/MapSettingsLayers.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { FC, ReactNode, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { updateLayersSettings } from 'reducers/map';
+import { GiElectric, GiUnplugged } from 'react-icons/gi';
+import { AiOutlineBlock } from 'react-icons/ai';
+import { MdSpaceBar } from 'react-icons/md';
+
+import BufferStopSVGFile from 'assets/pictures/layersicons/bufferstop.svg';
+import OPsSVGFile from 'assets/pictures/layersicons/ops.svg';
+import SwitchesSVGFile from 'assets/pictures/layersicons/switches.svg';
+import DetectorsSVGFile from 'assets/pictures/layersicons/detectors.svg';
+import SwitchSNCF, { SWITCH_TYPES } from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
+import { RootState } from 'reducers';
+
+type LayerSettings = RootState['map']['layersSettings'];
+
+interface FormatSwitchProps {
+  name: keyof Omit<LayerSettings, 'speedlimittag'>;
+  icon: ReactNode;
+  color?: string;
+  disabled?: boolean;
+}
+export const FormatSwitch: FC<FormatSwitchProps> = ({ name, icon, color, disabled }) => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation(['map-settings']);
+  const { layersSettings } = useSelector((state: RootState) => state.map);
+
+  const setLayerSettings = useCallback(
+    (setting: keyof LayerSettings) => {
+      dispatch(
+        updateLayersSettings({
+          ...layersSettings,
+          [setting]: !layersSettings[setting],
+        })
+      );
+    },
+    [dispatch]
+  );
```

Comment:
:warning: BUG: because of the callback, the switch works on the 1st click but not for the following clicks. 

A suggestion:
```suggestion
  const toggleLayerSetting = () => {
    dispatch(
      updateLayersSettings({
        ...layersSettings,
        [settingName]: !layersSettings[settingName],
      })
    );
  };
```

Reply:
this is not due to the useCallback, it just due to a missing dependency on the useCallback.
Can we re-enable the rule for the react exhaustive deps on hooks ?

Reply:
Yes we can I think, we should discuss this with the others, would you like to rise this issue on #osrd-front on slack ?